frontend/javascripts/oxalis/view/left-border-tabs/histogram_view.js

```diff
@@ -185,6 +191,79 @@ class Histogram extends React.PureComponent<HistogramProps, HistogramState> {
     500,
   );
 
+  getClippingValues = async (layerName: string, threshold: number = 0.05) => {
+    const { elementClass } = getLayerByName(Store.getState().dataset, layerName);
+    const [TypedArrayClass] = getConstructorForElementClass(elementClass);
+
+    const [cuboidXY, cuboidXZ, cuboidYZ] = await Promise.all([
+      api.data.getDataForViewport(OrthoViews.PLANE_XY, layerName),
+      api.data.getDataForViewport(OrthoViews.PLANE_XZ, layerName),
+      api.data.getDataForViewport(OrthoViews.PLANE_YZ, layerName),
+    ]);
+    const dataForAllViewports = new TypedArrayClass(
+      cuboidXY.length + cuboidXZ.length + cuboidYZ.length,
+    );
+
+    dataForAllViewports.set(cuboidXY);
+    dataForAllViewports.set(cuboidXZ, cuboidXY.length);
+    dataForAllViewports.set(cuboidYZ, cuboidXY.length + cuboidXZ.length);
+
+    const localHist = {};
+    for (let i = 0; i < dataForAllViewports.length; i++) {
+      if (dataForAllViewports[i] !== 0 && dataForAllViewports[i] in localHist) {
+        localHist[dataForAllViewports[i]] += 1;
+      } else {
+        localHist[dataForAllViewports[i]] = 1;
+      }
+    }
+
+    const accumulator = {};
+    let area = 0;
+    let lastKey = 0;
+    for (const key of Object.keys(localHist)) {
+      accumulator[key] = localHist[key] + area;
+      area += localHist[key];
+      lastKey = key;
+    }
+    const maximum = accumulator[lastKey];
+    const thresholdValue = (threshold * maximum) / 2.0;
+
+    let lowClip = -1;
+    for (const key of Object.keys(accumulator)) {
```

Comment:
the iteration method should be similar to my other comment when switching to Map().

Reply:
Awesome! Now I can properly expect <number, number> when iterating over the map. I think the code looks quite a bit better now 👍 Thanks!